### PR TITLE
gpuav: Fix VkGpuAssistedLayerTest.GpuBufferDeviceAddressOOB invalid SPIRV

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
Fixes #4983

**Review by ncesario-lunarg — APPROVED** [review]

**Review by spencer-lunarg — APPROVED** [review]

File: tests/vklayertests_gpu.cpp [hunk 1]
```diff
@@ -1043,137 +1044,181 @@ TEST_F(VkGpuAssistedLayerTest, GpuBufferDeviceAddressOOB) {
...
+        vk::CmdEndRenderPass(m_commandBuffer->handle());
+        m_commandBuffer->end();
+
+        for (const auto &test : testcases) {
+            auto *data = static_cast<VkDeviceAddress *>(buffer0.memory().map());
+            data[0] = test.push_constants[0];
+            data[1] = test.push_constants[1];
+            buffer0.memory().unmap();
+
+            if (test.error.size() > 0) {
```
Comment (arno-lunarg) [inline tests/vklayertests_gpu.cpp:1210]: Nit
```suggestion
            if (!test.error.empty()) {
```
(Same thing below)

File: tests/vklayertests_gpu.cpp [hunk 2]
```diff
@@ -1043,137 +1044,181 @@ TEST_F(VkGpuAssistedLayerTest, GpuBufferDeviceAddressOOB) {
...
+                m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "access out of bounds");
+            }
+            err = vk::QueueSubmit(m_device->m_queue, 1, &submit_info, VK_NULL_HANDLE);
+            ASSERT_VK_SUCCESS(err);
+            err = vk::QueueWaitIdle(m_device->m_queue);
+            ASSERT_VK_SUCCESS(err);
+            if (test.error.size() > 0) {
```
Comment (arno-lunarg) [inline tests/vklayertests_gpu.cpp:1217]: Nit
```suggestion
            if (!test.error.empty()) {
```

File: tests/vklayertests_gpu.cpp [hunk 3]
```diff
@@ -1043,137 +1044,181 @@ TEST_F(VkGpuAssistedLayerTest, GpuBufferDeviceAddressOOB) {
...
+            vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipe.handle());
+            vk::CmdPushConstants(m_commandBuffer->handle(), pipeline_layout.handle(), VK_SHADER_STAGE_VERTEX_BIT, 0,
+                                 sizeof(test.push_constants), test.push_constants);
+            vk::CmdSetViewport(m_commandBuffer->handle(), 0, 1, &viewport);
+            vk::CmdSetScissor(m_commandBuffer->handle(), 0, 1, &scissors);
+            vk::CmdDraw(m_commandBuffer->handle(), 3, 1, 0, 0);
+            vk::CmdEndRenderPass(m_commandBuffer->handle());
+            m_commandBuffer->end();
+
+            if (test.error.size() > 0) {
```
Comment (arno-lunarg) [inline tests/vklayertests_gpu.cpp:1137]: Nit
```suggestion
            if (!test.error.empty()) {
```

File: tests/vklayertests_gpu.cpp [hunk 4]
```diff
@@ -1184,55 +1229,54 @@ TEST_F(VkGpuAssistedLayerTest, GpuBufferDeviceAddressOOB) {
...
         mesh_pipe.AddDefaultColorAttachment();
-        err = mesh_pipe.CreateVKPipeline(mesh_pipeline_layout, renderPass());
+        err = mesh_pipe.CreateVKPipeline(mesh_pipeline_layout.handle(), renderPass());
         ASSERT_VK_SUCCESS(err);
         m_commandBuffer->begin(&begin_info);
         m_commandBuffer->BeginRenderPass(m_renderPassBeginInfo);
         vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, mesh_pipe.handle());
         VkDeviceAddress pushConstants[2] = {};
         pushConstants[0] = pBuffer;
         pushConstants[1] = 5;
```
Comment (arno-lunarg) [inline tests/vklayertests_gpu.cpp:1277]: Nit
```suggestion
        VkDeviceAddress pushConstants[2] = {pBuffer, 5};
```

**Review by arno-lunarg — COMMENTED** [review]
LGTM! Some nits

**Review by Tony-LunarG — APPROVED** [review]
LGTM - Nice test enhancements

File: tests/vklayertests_gpu.cpp (hunk 4 above)
Comment (jeremyg-lunarg) [reply tests/vklayertests_gpu.cpp:1277]: dang I missed this one

**Pull request merged**